scripts/utils.sh

```diff
@@ -43,3 +43,19 @@ append_nl_if_not(){
   fi
 }
 
+download_dataset() {
```

Comment:
I find this name confusing because it doesn't just download the dataset, but also installs dependencies and sources the setup. I would call this function `install_deps_source_setup` or something more descriptive like that :)